common/djangoapps/student/views.py

```diff
@@ -728,6 +732,17 @@ def dashboard(request):
 
     enterprise_message = get_dashboard_consent_notification(request, user, course_enrollments)
 
+    enterprise_customer = enterprise_customer_for_request(request)
+    if enterprise_customer:
+        consent_required_courses = frozenset(
```

Comment:
Is `frozenset` doing something special here? Could you use a [set comprehension](http://python-reference.readthedocs.io/en/latest/docs/comprehensions/set_comprehension.html) instead?

Reply:
Not sure specifically if we need a frozenset, but looked like all the other sets that were being passed to the context were using it, so I followed.

Reply:
fixed, use set comprehension now